fastfuels_core/onramps/hag_pim.py

```diff
@@ -0,0 +1,193 @@
+            )
+    if not abs(pim_res[0]) >= desired_res:
+        raise ValueError(
+                f"The resolution of the plot imputation map, {pim_res} is finer than the desired resolution, {desired_res}."
+            )
+
+def convert_to_cover(raster, min_value, desired_res):
+    '''
+    Returns: 
+        raster with values where the hag value is more than the min_value
+        - values = 1/(# of cells in one cell desired resolution raster)
+    '''
+    cover_raster = xr.where(raster > min_value, 1.0, 0.0) 
+    cover_raster.rio.write_nodata(0, inplace=True)
+    cover_raster.rio.write_crs(raster.rio.crs,
+                               inplace=True)
+    res = np.array(raster.rio.resolution())
+    res_scale = desired_res/abs(res)
+    cover_raster /= (res_scale[0]*res_scale[1])
+    return cover_raster
+
+def resample_raster(raster, desired_res):
+    '''
+    Returns: 
+        raster with desired resolution 
+        - if desired resolution is finer, the values 
+        are determined by the nearest previous cell
+        - if the desired resolution is more coarse, the 
+        values are determined by summing previous cells 
+    '''
+    res = np.array(raster.rio.resolution())
+    x_min = raster["x"].min().item() - abs(res[0]/2)
+    y_max = raster["y"].max().item() + abs(res[1]/2)
+    new_transform = from_origin(x_min, y_max, 
+                                desired_res, desired_res)
+    res_scale = abs(res)/desired_res
+    old_shape = np.array((raster.rio.height,
+                          raster.rio.width))
+    new_shape = old_shape*res_scale
+    new_shape = (int(new_shape[0]),
+                 int(new_shape[1]))
+    if res[0] < desired_res:
+        resample = Resampling.sum
+    else:
+        resample = Resampling.nearest
+    raster_resampled = raster.rio.reproject(
+        raster.rio.crs,
+        transform=new_transform,
+        shape=new_shape,
+        resampling=resample
+    )
+    raster_resampled = raster_resampled.fillna(0)
+    raster_resampled.rio.write_nodata(0, inplace=True)
+    return raster_resampled
+
+
+def interpolate_hag(hag_raster, pim_raster):
+    '''
+    Returns: 
+        hag raster with same coordinates as pim raster
+        - values were determined using nearest neighbor from hag input
+    '''
+    hag_interp = hag_raster.rio.reproject_match(pim_raster)
+    hag_interp.rio.write_crs(hag_raster.rio.crs, inplace=True)
+    hag_interp.rio.write_nodata(hag_raster.rio.nodata, inplace=True)
+    return hag_interp
+
+def interpolate_pim(pim_raster):
+    '''
+    Returns: 
+        plot imputation raster with all cells filled with data
+        - values were determined using nearest neighbor
+    '''
+    pim_interp = pim_raster.rio.interpolate_na(
+        method='nearest')
+    pim_interp.rio.write_crs(pim_raster.rio.crs, inplace=True)
+    pim_interp.rio.write_nodata(pim_raster.rio.nodata, inplace=True)
+    return pim_interp
```

Comment:
There's some potential here to combine these into a single interpolate_raster function that takes in the interpolation method, but I'd say that addressing this is low priority.

Reply:
Yeah, I have wondered about that too. Or we could just interpolate both within one function? I mean both of these steps are just using rasterio functions anyways, do we even really need these to be separate functions?

Reply:
I'll roll this into the next review.